### PR TITLE
Fix - RGB format for dashboard widget colours

### PR DESCRIPTION
**Pull request by RomainLvr**
## Checklist before requesting a review

*Please delete options that are not relevant.*

- [x] I have read the CONTRIBUTING document.
- [x] I have performed a self-review of my code.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [ ] This change requires a documentation update.

## Description

- It fixes !34379
- In the dashboard, the widget colours could not be in RGB format

## Screenshots (if appropriate):
image

image

### CODE REVIEW

File: src/Toolbox.php [hunk 1]
```diff
@@ -3439,6 +3439,18 @@ public static function getFgColor(string $color = "", int $offset = 40, bool $in
     {
         $fg_color = "FFFFFF";
         if ($color !== "") {
+            if (strpos($color, 'rgba') !== false || strpos($color, 'rgb') !== false) {
+                $rgb_color = [];
+                if (preg_match('/rgba?\((\d+),\s*(\d+),\s*(\d+),?\s*([\d\.]+)?\)/', $color, $matches)) {
```
Comment (trasher) [inline src/Toolbox.php:3444]: I guess the `preg_match` should be enough, no need for 2 checks.

**Comment by cconard96** [comment]
Dashboard cards were supposed to support transparency. Having a card at half transparency for example is a good way to make the cards look OK in both light and dark themes.

**Review by trasher — COMMENTED** [review]
Should not the alpha value be handled? This would need a test case I guess.

Appart of that, it seems correct from a code point of view, but I do not know for thun functional part. 
Also, did you see Curtis comment?

**Review by cedric-anne — APPROVED** [review]

**Review by cedric-anne — CHANGES_REQUESTED** [review]
My bad, if there is transparency set, then it should probably be handled. The lib we use to convert colors does not support it, but it should not be a pain to do it manually (the code to convert a number to its hex value is quiet simple).

**Review by cedric-anne — APPROVED** [review]

**Pull request merged**